Source/WebCore/platform/graphics/GCGLSpan.h

```diff
@@ -26,155 +26,9 @@
-        : GCGLSpan(array.data(), array.size())
-    { }
-    GCGLSpan(const ArrayBufferView& view)
-        : GCGLSpan(view.baseAddress(), view.byteLength())
-    { }
-    T* data() const noexcept { return m_data; }
-    size_t size() const noexcept { return m_bufSize; }
-    T& operator[](size_t i) { RELEASE_ASSERT(m_data && i < m_bufSize); return m_data[i]; }
-    T& operator*() { RELEASE_ASSERT(m_data && m_bufSize); return *m_data; }
-private:
-    T* m_data;
-    size_t m_bufSize;
-};
-
-template<>
-class GCGLSpan<GCGLvoid> {
-public:
-    GCGLSpan(GCGLvoid* array, size_t bufSize)
-        : m_data(array)
-        , m_bufSize(bufSize)
-    { }
-    template<typename U, size_t Sz>
-    GCGLSpan(U (&array)[Sz])
-        : GCGLSpan(array, Sz * sizeof(U))
-    { }
-    GCGLSpan(const GCGLSpan<GCGLvoid>& other)
-        : GCGLSpan(other.data(), other.size())
-    { }
-    template<typename U, size_t UExtent>
-    GCGLSpan(const GCGLSpan<U, UExtent>& other)
-        : GCGLSpan(other.data(), other.size() * sizeof(U))
-    { }
-    template<typename U, size_t inlineCapacity>
-    GCGLSpan(Vector<U, inlineCapacity>& array)
-        : GCGLSpan(array.data(), array.size() * sizeof(U))
-    { }
-    GCGLSpan(const ArrayBufferView& view)
-        : GCGLSpan(view.baseAddress(), view.byteLength())
-    { }
-    void* data() const noexcept { return m_data; }
-    size_t size() const noexcept { return m_bufSize; }
-private:
-    GCGLvoid* m_data;
-    size_t m_bufSize;
-};
-
-template<>
-class GCGLSpan<const GCGLvoid> {
-public:
-    GCGLSpan(const GCGLvoid* array, size_t bufSize)
-        : m_data(array)
-        , m_bufSize(bufSize)
-    { }
-    template<typename U, size_t Sz>
-    GCGLSpan(U (&array)[Sz])
-        : GCGLSpan(array, Sz * sizeof(U))
-    { }
-    GCGLSpan(const GCGLSpan<GCGLvoid>& other)
-        : GCGLSpan(other.data(), other.size())
-    { }
-    template<typename U, size_t UExtent>
-    GCGLSpan(const GCGLSpan<U, UExtent>& other)
-        : GCGLSpan(other.data(), other.size() * sizeof(U))
-    { }
-    GCGLSpan(const ArrayBufferView& view)
-        : GCGLSpan(view.baseAddress(), view.byteLength())
-    { }
-    const void* data() const noexcept { return m_data; }
-    size_t size() const noexcept { return m_bufSize; }
-private:
-    const GCGLvoid* m_data;
-    size_t m_bufSize;
-};
-
-template<typename T, size_t N>
-GCGLSpan(T (&)[N]) -> GCGLSpan<T, N>;
-
-template<typename T, size_t N>
-GCGLSpan(std::array<T, N>&) -> GCGLSpan<T, N>;
-
-template<typename T, size_t inlineCapacity>
-GCGLSpan(Vector<T, inlineCapacity>&) -> GCGLSpan<T>;
-
-template<typename T>
-GCGLSpan<T> makeGCGLSpan(T* data, size_t count)
-{
-    return GCGLSpan<T> { data, count };
-}
-
-template<size_t Extent, typename T>
-GCGLSpan<T, Extent> makeGCGLSpan(T* data)
-{
-    return GCGLSpan<T, Extent> { data };
-}
-
 template<typename... Types>
 struct GCGLSpanTuple {
```

Comment:
Should rename this file to `GCGLSpanTuple.h`

Reply:
Yeah. I was thinking maybe renaming it to WTF::SpanTuple, and thus to avoid churn leave it temporarily there in GCGLSpan.h